### PR TITLE
fix a few more warnings

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: winpr/libwinpr/sspi/Kerberos/kerberos.c [hunk 1]
```diff
@@ -659,117 +659,154 @@ static BOOL kerberos_mk_tgt_token(SecBuffer* buf, int msg_type, char* sname, cha
...
+				goto end;
+			free(lstr);
 		}
 
-		/* realm [3] Realm */
-		if (tag != 3 || !WinPrAsn1DecReadGeneralString(&dec2, &str))
-			goto fail;
-
-		*buf++ = '@';
-		strcpy(buf, str);
```
Comment (hardening) [inline winpr/libwinpr/sspi/Kerberos/kerberos.c:747]: Is it me or the goal of all these changes is just to not have a `strcpy` which seems correctly checked and used in this function ? At least the new code is less trivial to read...

File: winpr/libwinpr/sspi/Kerberos/kerberos.c [hunk 2]
```diff
@@ -659,117 +659,154 @@ static BOOL kerberos_mk_tgt_token(SecBuffer* buf, int msg_type, char* sname, cha
...
+	return ndst - dst;
+}
 
-	/* KERB-TGT-REQUEST (SEQUENCE) */
-	if (!WinPrAsn1DecReadSequence(&dec, &dec2))
+static BOOL kerberos_rd_tgt_req(WinPrAsn1Decoder* dec, char** target)
+{
+	BOOL rc = FALSE;
+	char* str = NULL;
+	if (!target)
```
Comment (hardening) [inline winpr/libwinpr/sspi/Kerberos/kerberos.c:674]: Check not needed this called only from this module, we control the callers.

Comment (akallabeth) [reply winpr/libwinpr/sspi/Kerberos/kerberos.c:674]: that is not quite correct.
the function is called with and without the parameter being `NULL`.
the decision is based on expectations (`if (context->state == KERBEROS_STATE_TGT_REQ && tok_id == TOK_ID_TGT_REQ)`) whereas the actual call is based on data read from the `ASN1` stream.

File: channels/tsmf/client/ffmpeg/tsmf_ffmpeg.c [hunk 3]
```diff
@@ -441,9 +441,12 @@ static BOOL tsmf_ffmpeg_decode_video(ITSMFDecoder* decoder, const BYTE* data, UI
 		av_image_fill_arrays(frame->data, frame->linesize, mdecoder->decoded_data,
 		                     mdecoder->codec_context->pix_fmt, mdecoder->codec_context->width,
 		                     mdecoder->codec_context->height, 1);
-		av_image_copy(frame->data, frame->linesize, (const uint8_t**)mdecoder->frame->data,
-		              mdecoder->frame->linesize, mdecoder->codec_context->pix_fmt,
-		              mdecoder->codec_context->width, mdecoder->codec_context->height);
+
+		const uint8_t** ptr =
+		    WINPR_REINTERPRET_CAST(mdecoder->frame->data, uint8_t**, const uint8_t**);
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply channels/tsmf/client/ffmpeg/tsmf_ffmpeg.c:446]: @mfleisz rechecking, but `char**` to `const char**` does not work well, maybe `const char* const*` (did not try that)

[the remark this answers is not in the document]
Comment (akallabeth) [reply channels/tsmf/client/ffmpeg/tsmf_ffmpeg.c:446]: @mfleisz ok, indeed, looks like I did check the wrong cast. reverting.

File: libfreerdp/codec/dsp_fdk_impl.c [hunk 4]
```diff
@@ -210,16 +210,31 @@ static void log_enc_info(const AACENC_InfoStruct* info, fdk_log_fkt_t log)
...
-		sprintf(tmp, "0x%02x", (int)info->confBuf[x]);
+		rc = snprintf(&confBuf[offset], remain - offset, "0x%02x", (int)info->confBuf[x]);
+		if (rc <= 0)
+			return;
+		offset += (size_t)rc;
 		if (x > 0)
-			strcat(confBuf, ", ");
-		strcat(confBuf, tmp);
+		{
+			rc = snprintf(&confBuf[offset], remain - offset, ", ");
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/codec/dsp_fdk_impl.c]: done

File: winpr/libwinpr/sspi/Kerberos/kerberos.c (hunk 2 above)
Comment (hardening) [reply winpr/libwinpr/sspi/Kerberos/kerberos.c:674]: correct I've been fooled by the diff view

File: winpr/libwinpr/sspi/Kerberos/kerberos.c (hunk 1 above)
Comment (hardening) [reply winpr/libwinpr/sspi/Kerberos/kerberos.c:747]: Well on this one, the job was correctly done in terms of checks. And I had to look at the previous version to understand what the new code was doing.

Comment (akallabeth) [reply winpr/libwinpr/sspi/Kerberos/kerberos.c]: @hardening ok, split out the changes in `kerberos.c` to a separate pr as that is indeed quite a change.

**Comment by akallabeth** [comment]
removed disputed changes, merging the rest.

**Pull request merged**